pkg/apis/core/v1alpha1/types_common.go

```diff
@@ -154,3 +154,33 @@ const (
 	// EventMigrationPreparationFailed indicates that Migration preparation failed.
 	EventMigrationPreparationFailed = "MigrationPreparationFailed"
 )
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// High Availability relevant types                                                             //
+//////////////////////////////////////////////////////////////////////////////////////////////////
+
+// HighAvailability specifies the configuration settings for high availability for a resource. Typical
+// usages could be to configure HA for shoot control plane or for seed system components.
+type HighAvailability struct {
+	// FailureTolerance holds information about failure tolerance level of a highly available resource.
+	FailureTolerance FailureTolerance `json:"failureTolerance" protobuf:"bytes,1,name=failureTolerance"`
+}
+
+// FailureTolerance describes information about failure tolerance level of a highly available resource.
+type FailureTolerance struct {
+	// FailureToleranceType specifies the type of failure that the highly available resource can tolerate
+	FailureToleranceType FailureToleranceType `json:"type" protobuf:"bytes,1,name=type"`
```

Comment:
```suggestion
	Type FailureToleranceType `json:"type" protobuf:"bytes,1,name=type"`
```
To make is synchronous with json/protobuf field.